scrapy/core/downloader/handlers/http11.py

```diff
@@ -415,7 +423,21 @@ def connectionLost(self, reason):
         body = self._bodybuf.getvalue()
         if reason.check(ResponseDone):
             self._finished.callback((self._txresponse, body, None))
-        elif reason.check(PotentialDataLoss):
+            return
+
+        if reason.check(PotentialDataLoss):
             self._finished.callback((self._txresponse, body, ['partial']))
-        else:
-            self._finished.errback(reason)
+            return
+
+        if reason.check(ResponseFailed) and any(r.check(_DataLoss) for r in reason.value.reasons):
+            if self._allow_dataloss:
+                self._finished.callback((self._txresponse, body, ['partial']))
+                return
+
+            elif not self._allow_dataloss_hinted:
+                logger.warn("Found data loss in %s. If you want to process broken "
+                            "responses set the setting DOWNLOAD_ALLOW_DATALOSS = True",
```

Comment:
I think it worths adding a note that the message won't be shown for further requests, like we do in dupefilters.